examples/getting_started/quickstart.ipynb

```diff
@@ -0,0 +1,626 @@
+    "\n",
+    "Here's an example of how to define and use a custom tool, continuing with our capital city theme:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from mirascope.core import BaseTool, openai, prompt_template\n",
+    "\n",
+    "\n",
+    "class FormatCapital(BaseTool):\n",
+    "    city: str\n",
+    "    country: str\n",
+    "\n",
+    "    def call(self) -> str:\n",
+    "        return f\"The capital of {self.country} is {self.city}.\"\n",
+    "\n",
+    "\n",
+    "@openai.call(\"gpt-4o-mini\", tools=[FormatCapital])\n",
+    "@prompt_template(\"What is the capital of {country}?\")\n",
+    "def get_capital(country: str): ...\n",
+    "\n",
+    "\n",
+    "response = get_capital(\"France\")\n",
+    "if tools := response.tools:\n",
+    "    for tool in tools:\n",
+    "        print(tool.call())\n",
+    "else:\n",
+    "    print(response.content)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's break down this example:\n",
+    "\n",
+    "1. We define a custom tool `FormatCapital` that inherits from `BaseTool`. This tool takes a city and country, and formats them into a sentence stating the capital.\n",
+    "\n",
+    "2. The `FormatCapital` class has two attributes (`city` and `country`) and a `call` method that returns the formatted string.\n",
+    "\n",
+    "3. In the `get_capital` function, we use the `@openai.call` decorator and include our `FormatCapital` tool in the `tools` parameter.\n",
+    "\n",
+    "4. The LLM can now use this tool when generating its response. If it decides to use the tool, it will provide values for `city` and `country`.\n",
+    "\n",
+    "5. When we call `get_capital`, we check if any tools were used in the response. If so, we call each tool and print its output. Otherwise, we print the direct content of the response.\n",
+    "\n",
+    "This approach allows the LLM to generate structured data (in this case, a capital city and its country) in a format that's easy for your application to process.\n",
+    "\n",
+    "Tools can be used for various purposes, such as:\n",
+    "- Formatting output in a specific way\n",
+    "- Performing calculations (e.g., population density given area and population)\n",
+    "- Accessing external data sources (e.g., fetching up-to-date information about a city)\n",
+    "- Implementing custom logic that the LLM can leverage (e.g., determining the continent based on the country)\n",
+    "\n",
+    "By providing tools, you give the LLM additional capabilities that it can use to generate more accurate, formatted, or context-aware responses. In this case, the `FormatCapital` tool ensures that the capital information is always presented in a consistent format."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Asynchronous Processing\n",
```

Comment:
I think we should also show async streaming at least in this section?

another option for all of these sections if we want to keep them brief would be to add a link to their respective learn docs that contains all of the information?